jax/numpy/linalg.py

```diff
@@ -148,12 +148,88 @@ def _slogdet_jvp(primals, tangents):
   return (sign, ans), (sign_dot, ans_dot)
 
 
+def _cofactor_solve(a, b):
+  """Equivalent to det(a)*solve(a, b) for nonsingular mat.
+
+  Intermediate function used for jvp and vjp of det.
+  This function borrows heavily from jax.numpy.linalg.solve and
+  jax.numpy.linalg.slogdet to compute the gradient of the determinant
+  in a way that is well defined even for low rank matrices.
+  * assumes a is at least rank n-1
+  * assumes u_{nn} is the element set to 0 in rank n-1 cases.
+
+  Args:
+    a: A square matrix or batch of matrices, possibly singular.
+    b: A matrix, or batch of matrices of the same dimension as a.
+
+  Returns:
+    det(a) and cofactor(a)^T*b, aka adjugate(a)*b
```

Comment:
do you have a reference on the method you use here for calculating the adjugate?

Reply:
Nope. Derived it myself. I could add a short description to the docstring?

Reply:
Done.